lib/ansible/playbook/role/__init__.py

```diff
@@ -317,6 +317,12 @@ def _prepend_validation_task(self, task_data, argspecs):
             # Determine the role entry point so we can retrieve the correct argument spec.
             # This comes from the `tasks_from` value to include_role or import_role.
             entrypoint = self._from_files.get('tasks', 'main')
+
+            # Remove the file extension
+            i = entrypoint.rfind('.')
```

Comment:
or just use splitext and compare the first item ?

Reply:
The other concern here is pathing.  In the original report, `tasks_from` ended up as an absolute path.

Reply:
the path should always be rolepath/tasks/ (we dont allow subdirs in _from), so can use basename or just capture the value before it gets transformed

Reply:
he, just tested and putting the abs path in `tasks_from ` does work, so you really cannot rely on it not being a path at one point

Reply:
`tasks_from: tasks/stuff.yml` .. also works

Reply:
> The other concern here is pathing. In the original report, tasks_from ended up as an absolute path.

So we essentially need the [stem](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem). I'll work on that, and add some tests for these cases.

Reply:
cannot use pathlib until we split controller tests ..

Reply:
Yup, I just really like a lot of the things it provides and it's good for inspiration.

Reply:
I think I have a better solution now.. Thanks for the feedback.